### PR TITLE
[NFC] Add InputObjects struct wrapper

### PR DESCRIPTION
**Pull request by lxfind**
Merge a bunch of input object related functions into a wrapper.

### CODE REVIEW

File: crates/sui-core/src/transaction_input_checker.rs [hunk 1]
```diff
@@ -17,12 +18,94 @@ use tracing::{debug, instrument};
...
+                    } else {
+                        Some(*object_ref)
+                    }
+                }
+                InputObjectKind::SharedMoveObject(_) => None,
+            })
+            .collect();
+
+        debug!(
+            num_mutable_objects = owned_objects.len(),
```
Comment (sblackshear) [inline crates/sui-core/src/transaction_input_checker.rs:56]: This looks left over from elsewhere--I don't see where `num_mutable_objects` is bound.

File: crates/sui-core/src/transaction_input_checker.rs [hunk 2]
```diff
@@ -17,12 +18,94 @@ use tracing::{debug, instrument};
...
+    }
+
+    pub fn transaction_dependencies(&self) -> BTreeSet<TransactionDigest> {
+        self.objects
+            .iter()
+            .map(|(_, obj)| obj.previous_transaction)
+            .collect()
+    }
+
+    pub fn active_inputs(&self) -> Vec<ObjectRef> {
```
Comment (sblackshear) [inline crates/sui-core/src/transaction_input_checker.rs:78]: Wondering if `mutable_inputs` might be clearer (unless we use the term "active" elsewhere).

**Review by sblackshear — APPROVED** [review]
Nice simplification!

File: crates/sui-core/src/transaction_input_checker.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui-core/src/transaction_input_checker.rs:56]: Ha, so this is confusing but `debug!` is a tracing macro that dumps a debug log. The syntax `num_mutable_objects = owned_objects.len()` adds an indexable metric/key to the log named `num_mutable_objects` with the value of `owned_objects.len()`

**Pull request merged**